### PR TITLE
UefiPayloadPkg: remove LAN ROMs and add external ROM integration

### PR DESCRIPTION
**Pull request by miczyg1**
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

### CODE REVIEW

**Comment by pietrushnic** [comment]
@miczyg1 it is not that simple, you have to rebase and get rid of binaries in history. I already did part of that work on this branch: https://github.com/Dasharo/edk2/tree/explain-ipxe-build-next

Question is if I didn't messed up since I amended commits (some probably should be removed anyway). Please test before use.

**Comment by macpijan** [comment]
Michal's concern was whether we want to break previous (not so old) releases right now with purging history.

File: UefiPayloadPkg/UefiPayloadPkg.fdf [hunk 1]
```diff
@@ -233,44 +233,17 @@ INF MdeModulePkg/Bus/Usb/UsbMassStorageDxe/UsbMassStorageDxe.inf
...
-# Available at https://downloadcenter.intel.com/download/29137?v=t
 #
-!if $(NETWORK_ENABLE)
-!if $(ARCH) == IA32
-  # FIXME
-!else
-!if $(NETWORK_INTEL_PRO2500)
+!if $(NETWORK_LAN_ROM) == TRUE
+#
+# Build system must place the LanRom.efi before buildnig the UEFI Payload.
```
Comment (mkopec) [inline UefiPayloadPkg/UefiPayloadPkg.fdf]:
```suggestion
# Build system must place the LanRom.efi before building the UEFI Payload.

```

**Review by mkopec — COMMENTED** [review]
Just one nitpick

**Comment by miczyg1** [comment]
@mkopec done

**Pull request merged**